include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,298 @@
+  struct options : Settings
+  {
+    template <rbr::concepts::option... Options>
+    constexpr EVE_FORCEINLINE explicit options(Options && ... opts) : Settings(EVE_FWD(opts) ...) {}
+
+    template <typename... Options>
+    constexpr EVE_FORCEINLINE explicit options(rbr::settings<Options...> const& opts) : Settings(opts) {}
+  };
+
+  template <rbr::concepts::option ... Options>
+  options(Options&& ... opts) -> options<decltype(rbr::settings(EVE_FWD(opts) ...))>;
+
+  template <typename... Options>
+  options(rbr::settings<Options...> const&)  -> options<rbr::settings<Options...>>;
+}
+
+//======================================================================================================================
+// Overload's helpers
+//======================================================================================================================
+namespace eve::detail
+{
+  //====================================================================================================================
+  //  Delay the access to defaults() to after the proper operator() has been selected.
+  //  Also returns an empty set of options by default
+  //====================================================================================================================
+  template<typename T, typename...> EVE_FORCEINLINE constexpr auto defaults()
+  {
+    if constexpr(requires { T::defaults(); }) return T::defaults();
+    else                                      return options{};
+  }
+
+  //====================================================================================================================
+  // Internal concept checking if an option is supported by a specification
+  //====================================================================================================================
+  template<typename Spec, typename O>
+  concept supports_option = requires(Spec sp, O const& o) { { sp[o] }; };
+
+  //====================================================================================================================
+  // Internal concept checking if an option is supported by a any specification in a given subset
+  //====================================================================================================================
+  template<typename O, typename... Specs>
+  concept is_supported = (supports_option<Specs,O> || ... );
+
+  //====================================================================================================================
+  // Internal option carrying conditional mask or conditional expressions
+  //====================================================================================================================
+  struct condition_key_t : rbr::as_keyword<condition_key_t>
+  {
+    using rbr::as_keyword<condition_key_t>::operator=;
+    template<typename T> static constexpr bool check()
+    {
+      return conditional_expr<std::remove_cvref_t<T>> || logical_value<std::remove_cvref_t<T>>;
```

Comment:
this seems unnessesary. The `process_option` has to do the checks

Reply:
no. That is not related to that.
Raberu asks for option level check to prevent key = bullshti to occurs.
Now, we indeed never let ppl use condition_key, we cna maybe remove it but it is unrelated, I need you to understadn this point.

Reply:
I understand. We just need a compile time map, that's all we use from raberu. Otherwise our usecase is different. This doesn't do anything

Reply:
Sure